frontend/src/views/Lobby.tsx

```diff
@@ -432,17 +441,38 @@ function LobbyPage() {
       })
       .finally(() => {
         setLoading(false);
+        setTempSelectedProblems([]);
+        setModifiedProblems(false);
       });
   };
 
+  // Add a problem to be selected - these will be submitted as a batch.
   const addProblem = (newProblem: SelectableProblem) => {
-    const newProblems = [...selectedProblems, newProblem];
-    updateSelectedProblems(newProblems);
+    const problemsToUse = modifiedProblems ? tempSelectedProblems : selectedProblems;
+    const newProblems = [...problemsToUse, newProblem];
+    setTempSelectedProblems(newProblems);
+    setModifiedProblems(true);
   };
 
-  const removeProblem = (index: number) => {
-    const newProblems = selectedProblems.filter((_, i) => i !== index);
-    updateSelectedProblems(newProblems);
+  // Submit the batch of selected problems and close the modal.
+  const submitTempProblems = () => {
+    updateSelectedProblems(tempSelectedProblems);
+    setShowProblemSelector(false);
+  };
```

Comment:
Should we set modified problems here, or is that not needed for some reason?

Reply:
It should be called in the `updateSelectedProblems` function once the POST request has gone through, I think it was done that way to prevent it from clearing the selection and showing the old problems while the request is loading